custom/icds_reports/reports/disha.py

```diff
@@ -81,32 +81,32 @@ def _write_data_in_chunks(self, temp_path):
         #   }
         columns = self._get_columns()
         indicators = self._get_rows()
-        metadata_line = '{{'\
+        metadata_line = u'{{'\
             '"month":"{month}", '\
             '"state_name": "{state_name}", '\
             '"column_names": {columns}, '\
             '"rows": ['.format(
                 month=self.month,
                 state_name=self.state_name,
-                columns=json.dumps(columns, ensure_ascii=False))
-        with open(temp_path, 'w', encoding='utf-8') as f:
-            f.write(metadata_line)
-            written_count = 0
-            num_batches = 10
-            for count, (_, end, total, chunk) in enumerate(batch_qs(indicators, num_batches=num_batches)):
-                chunk_string = json.dumps(list(chunk), ensure_ascii=False)
-                # chunk is list of lists, so skip enclosing brackets
-                f.write(chunk_string[1:-1])
-                written_count += len(chunk)
-                if written_count != total:
-                    f.write(",")
-                logger.info("Processed {count}/{total} batches".format(count=count, total=num_batches))
-            f.write("]}")
+                columns=json.dumps(columns, ensure_ascii=False)).encode('utf8')
+        file_obj.write(metadata_line)
+        written_count = 0
+        num_batches = 10
+        for count, (_, end, total, chunk) in enumerate(batch_qs(indicators, num_batches=num_batches)):
+            chunk_string = json.dumps(list(chunk), ensure_ascii=False).encode('utf8')
+            # chunk is list of lists, so skip enclosing brackets
+            file_obj.write(chunk_string[1:-1])
+            written_count += len(chunk)
+            if written_count != total:
+                file_obj.write(",")
+            logger.info("Processed {count}/{total} batches".format(count=count, total=num_batches))
+        file_obj.write("]}")
 
     def build_export_json(self):
         with TransientTempfile() as temp_path:
-            self._write_data_in_chunks(temp_path)
-            with open(temp_path, 'r', encoding='utf-8') as f:
```

Comment:
This was definitely wrong. Up to you if you want to keep the change of reading/writing from the same file object, but you could revert the changes here and simply change this to
```py
with open(temp_path, 'rb') as f:
```